data_safe_haven/config/backend_settings.py

```
@@ -0,0 +1,144 @@
+        if location:
+            self.logger.debug(f"Updating '[green]{location}[/]' to '{location}'.")
+            self._location = location
+        if name:
+            self.logger.debug(f"Updating '[green]{name}[/]' to '{name}'.")
+            self._name = name
+        if subscription_name:
+            self.logger.debug(
+                f"Updating '[green]{subscription_name}[/]' to '{subscription_name}'."
+            )
+            self._subscription_name = subscription_name
+
+        # Write backend settings to disk (this will trigger errors for uninitialised parameters)
+        self.write()
+
+    @property
+    def admin_group_id(self) -> str:
+        if not self._admin_group_id:
+            raise DataSafeHavenParameterException(
+                "Azure administrator group not provided: use '[bright_cyan]--admin-group[/]' / '[green]-a[/]' to do so."
+            )
+        return self._admin_group_id
+
+    @property
+    def location(self) -> str:
+        if not self._location:
+            raise DataSafeHavenParameterException(
+                "Azure location not provided: use '[bright_cyan]--location[/]' / '[green]-l[/]' to do so."
+            )
+        return self._location
+
+    @property
+    def name(self) -> str:
+        if not self._name:
+            raise DataSafeHavenParameterException(
+                "Data Safe Haven deployment name not provided: use '[bright_cyan]--deployment-name[/]' / '[green]-d[/]' to do so."
+            )
+        return self._name
+
+    @property
+    def subscription_name(self) -> str:
+        if not self._subscription_name:
+            raise DataSafeHavenParameterException(
+                "Azure subscription not provided: use '[bright_cyan]--subscription[/]' / '[green]-s[/]' to do so."
+            )
+        return self._subscription_name
+
+    def read(self) -> None:
+        """Read settings from YAML file"""
+        if self.config_file_path.exists():
```

Do we want to raise an exception if it does not exist?

---

No. The logic is:

1. Try to read variables from YAML file
2. Overwrite using any variables provided at the command line
3. Write to YAML file (which raises exceptions if any variables are missing)